File: src/main/java/com/todoist/model/Sanitizers.java
package com.todoist.model;

import java.util.regex.Pattern;

public class Sanitizers {
    public static final String PROJECT_NAME_INVALID_CHARACTERS = "<>\"=#+";
    public static final String LABEL_NAME_INVALID_CHARACTERS = "<>\"=#+\\s%!?~:@";
    public static final String FILTER_NAME_INVALID_CHARACTERS = "<>\"=#+";

    public static final Pattern PROJECT_NAME_INVALID_PATTERN =
            Pattern.compile("[" + PROJECT_NAME_INVALID_CHARACTERS + "]+");
    public static final Pattern LABEL_NAME_INVALID_PATTERN =
            Pattern.compile("[" + LABEL_NAME_INVALID_CHARACTERS + "]+");
    public static final Pattern FILTER_NAME_INVALID_PATTERN =
            Pattern.compile("[" + FILTER_NAME_INVALID_CHARACTERS + "]+");

    public static final String REPLACEMENT = "_";
}

File: src/main/java/com/todoist/pojo/Sanitizers.kt
package com.todoist.pojo

import java.util.regex.Pattern

object Sanitizers {
    const val WORKSPACE_NAME_INVALID_CHARACTERS = "[#\"()|&!,]+"
    const val PROJECT_NAME_INVALID_CHARACTERS = "#\"()|&!,"
    const val SECTION_NAME_INVALID_CHARACTERS = "/\"()|&!,"
    const val LABEL_NAME_INVALID_CHARACTERS = "@\"\\s()|&!,"

    @JvmField
    val WORKSPACE_NAME_INVALID_PATTERN: Pattern =
        Pattern.compile("[$WORKSPACE_NAME_INVALID_CHARACTERS]+")

    @JvmField
    val PROJECT_NAME_INVALID_PATTERN: Pattern =
        Pattern.compile("[$PROJECT_NAME_INVALID_CHARACTERS]+")

    @JvmField
    val SECTION_NAME_INVALID_PATTERN: Pattern =
        Pattern.compile("[$SECTION_NAME_INVALID_CHARACTERS]+")

    @JvmField
    val LABEL_NAME_INVALID_PATTERN: Pattern = Pattern.compile("[$LABEL_NAME_INVALID_CHARACTERS]+")

    const val REPLACEMENT = "_"
}
